docs/Contributing/API-for-contributors.md

```diff
@@ -2680,6 +2681,30 @@ Sends a signal to Fleet Desktop to initiate a FileVault key escrow. This is usef
 
 `Status: 204`
 
+
+### Report an agent error
+
+Notifies the server about an agent error, resulting in two outcomes:
+
+- The error gets saved in Redis and can later be accessed using `fleetctl debug archive`.
+- The server consistently replies with a `500` status code, which can serve as a signal to activate an alarm through a monitoring tool.
```

Comment:
Thinking out loud, is it a bit abusing the HTTP standard for alerting considerations? We wouldn't be able to tell if the request itself was successful or not (i.e. if the request to notify about an error succeeds, it would logically return 200, but here we make it return 500 even if it succeeds, but what if it failed saving the error, e.g. redis is down?).

Reply:
It's a good point, we haven't really discussed, as it was specified this way (see https://github.com/fleetdm/fleet/issues/13238#issuecomment-1671769460)

IIRC from the estimation session, the idea is to save #g-infra some work and support a particular customer request